rmgpy/molecule/molecule.py

```diff
@@ -1595,4 +1595,38 @@ def toGroup(self):
         
         return group
 
-
+    def getNthNeighbor(self, startingAtoms, n, blacklist):
+        '''
+        Recursively get the Nth nonHydrogen neighbors of the startingAtoms.
+        StartingAtoms should be a list.
+        Return a list.
+        '''
+        neighbors = []
+        for atom in startingAtoms:
+            for neighbor in self.getBonds(atom):
```

Comment:
I ran some tests out of interest, which you can see [here](https://github.com/mliu49/RMG-stuff/blob/master/Sandbox.ipynb). All three versions of this loop were timed, taking the cumulative time for 100,000 calls, repeated 5 times, then averaged.

I agree that speed is probably not an issue for this method, so it doesn't really matter too much which way you choose. However, the current method (adding all atoms, then removing non-hydrogens) is marginally slower than only adding non-hydrogens in the first place. Whether you use two for loops or list comprehension doesn't make that much of a difference.